pyshelf/search/manager.py

```diff
@@ -0,0 +1,140 @@
+from elasticsearch_dsl.query import Q
+from elasticsearch_dsl import Search
+from pyshelf.search.type import Type as SearchType
+from pyshelf.search.sort_type import SortType
+from pyshelf.search.metadata import Metadata
+from distutils.version import LooseVersion
+
+
+class Manager(object):
+    def __init__(self, search_container):
+        self.search_container = search_container
+        self.tilde_search = None
+
+    def search(self, criteria, key_list=None):
+        """
+            Builds ElasticSearch query.
+
+            Args:
+                criteria(dict): Criteria to use to initiate search. Example below.
+                key_list(list): List of keys to receive back from a search.
+
+            Returns:
+                dict: each element in the outer dict represents a search "hit"
+                      with the returned keys specified in key_list.
+
+        Example of criteria:
+
+            {
+                "search": [
+                    {
+                        "field": "version",
+                        "value": "1.1",
+                        "search_type": SearchType.TILDE
+                    }
+                ],
+                "sort": [
+                    {
+                        "field": "version",
+                        "sort_type": SortType.VERSION,
+                        "flag_list": [
+                            SortType.ASC
+                        ]
+                    }
+                ]
+            }
+        """
+        query = Search().index(Metadata._doc_type.index).query(self._build_query(criteria["search"]))
+        results = query.execute()
+        filtered_results = self._filter_results(results.hits, key_list)
+        formated_results = self._sort_results(filtered_results, criteria.get("sort"))
+
+        return formated_results
+
+    def _sort_results(self, filtered_results, sort_criteria=None):
+        """
+            Sorts results based on sort criteria.
+
+            Args:
+                filtered_results(list(dict)): list of dictionaries representing hits from Elasticsearch.
+
+            Returns:
+                list(dict): Sorts the filtered results that are passed in.
+        """
+        if not sort_criteria:
+            return filtered_results
+
+        for criteria in sort_criteria:
```

Comment:
We talked about this, but the order these sorts are applied in need to be in the opposite order.